modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2675,15 +2675,17 @@ def broadcast_apply_full_axis(
             enumerate_partitions=enumerate_partitions,
             keep_partitioning=keep_partitioning,
         )
-        # Index objects for new object creation. This is shorter than if..else
-        kw = self.__make_init_labels_args(new_partitions, new_index, new_columns)
+        kw = {}
         if dtypes == "copy":
             kw["dtypes"] = self._dtypes
-        elif dtypes is not None:
+        elif dtypes is not None and new_columns is not None:
+            columns = new_columns if new_columns is not None else self.columns
             kw["dtypes"] = pandas.Series(
-                [np.dtype(dtypes)] * len(kw["columns"]), index=kw["columns"]
+                [np.dtype(dtypes)] * len(columns), index=columns
             )
-        result = self.__constructor__(new_partitions, **kw)
+        result = self.__constructor__(
+            new_partitions, index=new_index, columns=new_columns, **kw
```

Comment:
Now there might be a case when all the optional kwargs are None. This may speep up certain cases but slow down others. I would like to look at perf measurements on where we see speed up and we don't

Reply:
They will also be calculated using the same function `_compute_axis_labels_and_lengths` if necessary. We are simply postponing the moment of calculation. What kind of slowdown are you talking about?

Reply:
The change may speed up pivot_table and probably other stuff that calls `broadcast_apply_full_axis` as a single call but subsequent operations may slow down. I would like to see some measurents for a chain of operations (pivot_table + other call that requires the metadata to be materialized).

Reply:
We should probably implement `__make_init_labels_args` which would be engine specific and computes metadata in remote calls giving object refs / futures. That is what we discussed offline on the meeting. Could you please create an issue for that?

Reply:
> The change may speed up pivot_table and probably other stuff that calls `broadcast_apply_full_axis` as a single call but subsequent operations may slow down. I would like to see some measurents for a chain of operations (pivot_table + other call that requires the metadata to be materialized).

This is true for each operation in the sequence individually, but not for the sequence as a whole.

Reply:
> __make_init_labels_args

https://github.com/modin-project/modin/issues/5574

Reply:
> > The change may speed up pivot_table and probably other stuff that calls `broadcast_apply_full_axis` as a single call but subsequent operations may slow down. I would like to see some measurents for a chain of operations (pivot_table + other call that requires the metadata to be materialized).
> 
> This is true for each operation in the sequence individually, but not for the sequence as a whole.

Can you provide an example (time for a sequence of operations, as well as time for each single operation in the sequence)?

Reply:
> LGTM but I still want to see some numbers.

I thought I already answered, but no. Yes, I will.